NEWS.md

```diff
@@ -4,6 +4,10 @@
 
 # data.table [v1.12.9](https://github.com/Rdatatable/data.table/milestone/19)  (in development)
 
+## POTENTIALLY BREAKING CHANGES
+
+1. `fread` now supports native parsing of `%Y-%m-%d`, `%Y-%m-%d %H:%M:%OS`, and [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) `%Y-%m-%dT%H:%M:%OS%z`. Detected dates are returned as `data.table`'s `integer`-backed `IDate` class (see `?IDate`), and times are returned as `POSIXct` columns with time zone set to `UTC` (with offset indicated by `%z` applied). To display this column in the time zone of your choosing, recall that time zones in `POSIXct` vectors are just printing artifacts -- all `POSIXct` columns use numeric seconds since 1970-01-01 00:00:00 UTC as the underlying representation, and the time zone is only applied when `print` (or `format`) is invoked. As such, changing time zones can be done near instantaneously & with no copies by `setattr(tzcol, 'tzone', 'Asia/Yangon')` (or any other time zone available in `OlsonNames()`). Since this is technically a breaking change (i.e. existing code may depend on dates and times being read as character), an option is provided to restore the old behaviour (`options(datatable.old.fread.datetime.character=TRUE)`). Or, you may wish to simply downgrade to the old version until you can update your `fread` calls. In addition to convenience, `fread` is now significantly faster in the presence of date and time columns. Since this is a potentially breaking change, the minor version number is bumped: v1.13.0.
```

Comment:
maybe we could set `datatable.old.fread.datetime.character=TRUE` as default and release as 1.12.10, and then after release change this default to `FALSE`?

Reply:
This will also help users to prepare for migration if we mention now that this will be a breaking change from the next release. Users can now upgrade their code during 1.12.10 cycle, and not being stuck on 1.12.8 before they upgrade.

Reply:
My gut feel is that would be taking on too much work (releasing to CRAN is a lot of work) due to people who can't just downgrade or set an option. The same people who don't read NEWS. Why do the majority have to have the inconvenience of having to set an option for something that is long-overdue and much needed, just because the minority upgrade production systems directly without testing first, find it too hard to downgrade, and won't even read the first item of the NEWS file to find out the option to set.
There will be problems, bugs and changes to defaults. By nudging straight away, though, we get through those faster. It's a .0 release (to convey be-aware) and the NEWS item is right up at the top in it's own section in capitals. And the change is long-overdue, much needed, and compelling.

Reply:
CRAN revdeps results may well make the decision for us. If a lot of packages fail due to this change, we might have to give those packages more time by having this new feature off by default. The difference with CRAN packages are that checks are run with latest version and they don't have any choice about that.  Out in the wild is different because users have a choice to upgrade or not, downgrade, or use the new option.